### PR TITLE
HUSH-74 Add snoopy helm chart

### PR DESCRIPTION
**Pull request by r-bk**
Add initial version of snoopy's helm chart `hush-sensor` + basic CI.

### CODE REVIEW

**Comment by r-bk** [comment]
Revision 1
---
**Compare**
- [0..1](https://github.com/hushsecurity/helm-charts/compare/0d4d38026f6abf6f44a1d5885b5e7a13cc657683..2590ffca7cf52e4a9c017314a440eb2967f01661)

**CC** @sladkani @horomena


[//]: # (gh-pr-revision:{"number":1,"hash":"2590ffca7cf52e4a9c017314a440eb2967f01661","baseHash":"0d4d38026f6abf6f44a1d5885b5e7a13cc657683"})

**Comment by r-bk** [comment]
Revision 2
---


[//]: # (gh-pr-revision-body-start)

added schema validation with 

[//]: # (gh-pr-revision-body-end)

**Compare**
- [1..2](https://github.com/hushsecurity/helm-charts/compare/2590ffca7cf52e4a9c017314a440eb2967f01661..3b430b9ddc3bfc91ed394ab653ef310738d27c29)
- [0..1](https://github.com/hushsecurity/helm-charts/compare/0d4d38026f6abf6f44a1d5885b5e7a13cc657683..2590ffca7cf52e4a9c017314a440eb2967f01661), [0..2](https://github.com/hushsecurity/helm-charts/compare/0d4d38026f6abf6f44a1d5885b5e7a13cc657683..3b430b9ddc3bfc91ed394ab653ef310738d27c29)

**CC** @sladkani @horomena


[//]: # (gh-pr-revision:{"number":2,"hash":"3b430b9ddc3bfc91ed394ab653ef310738d27c29","baseHash":"0d4d38026f6abf6f44a1d5885b5e7a13cc657683"})

**Review by horomena — APPROVED** [review]
Looks good.
The only thing I would check is whether we need to create some hooking point for easier whitelist/blacklist by node/namespace or he can already do all this easily from outside without a hooking point.

**Comment by sladkani** [comment]
lets have a good default for the namespace for which we deploy (and have it override-able).  we'll create the Namespace object if its not same as .Release.Namespce

File: charts/hush-sensor/templates/daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      volumes:
+        - name: containerd
+          hostPath:
+            path: /run/containerd
+            type: Directory
+        - name: cgroupfs
+          hostPath:
+            path: /sys/fs/cgroup
+            type: Directory
+        - name: docker-socket
```
Comment (sladkani) [inline charts/hush-sensor/templates/daemonset.yaml:46]: no need for "docker-socket" volume in k8s env

Comment (r-bk) [reply charts/hush-sensor/templates/daemonset.yaml]: removed

**Comment by r-bk** [comment]
> also, you should provide snoopy the actual namespace being used, so snoopy can filter it (to avoid trace itself).

@sladkani I will implement this as part of HUSH-269

**Comment by r-bk** [comment]
Revision 3
---


[//]: # (gh-pr-revision-body-start)

Fixed small issues that prevented deployment in EKS.
Will add Namespace in a separate revision.


[//]: # (gh-pr-revision-body-end)

**Compare**
- [2..3](https://github.com/hushsecurity/helm-charts/compare/3b430b9ddc3bfc91ed394ab653ef310738d27c29..553a340ce353f4e1b50c56080dc976c905cf8fd9)
- [1..2](https://github.com/hushsecurity/helm-charts/compare/2590ffca7cf52e4a9c017314a440eb2967f01661..3b430b9ddc3bfc91ed394ab653ef310738d27c29), [1..3](https://github.com/hushsecurity/helm-charts/compare/2590ffca7cf52e4a9c017314a440eb2967f01661..553a340ce353f4e1b50c56080dc976c905cf8fd9)
- [0..1](https://github.com/hushsecurity/helm-charts/compare/0d4d38026f6abf6f44a1d5885b5e7a13cc657683..2590ffca7cf52e4a9c017314a440eb2967f01661), [0..3](https://github.com/hushsecurity/helm-charts/compare/0d4d38026f6abf6f44a1d5885b5e7a13cc657683..553a340ce353f4e1b50c56080dc976c905cf8fd9)



[//]: # (gh-pr-revision:{"number":3,"hash":"553a340ce353f4e1b50c56080dc976c905cf8fd9","baseHash":"0d4d38026f6abf6f44a1d5885b5e7a13cc657683"})

**Comment by r-bk** [comment]
Revision 4
---


[//]: # (gh-pr-revision-body-start)

1. Add `Namespace` object with default name "hush-security"
2. Put all resourcess in the namespace specified in `values.yaml`.



[//]: # (gh-pr-revision-body-end)

**Compare**
- [3..4](https://github.com/hushsecurity/helm-charts/compare/553a340ce353f4e1b50c56080dc976c905cf8fd9..25a1a5ba95d97e7da0459964464fb9b99a18badf)
- [2..3](https://github.com/hushsecurity/helm-charts/compare/3b430b9ddc3bfc91ed394ab653ef310738d27c29..553a340ce353f4e1b50c56080dc976c905cf8fd9), [2..4](https://github.com/hushsecurity/helm-charts/compare/3b430b9ddc3bfc91ed394ab653ef310738d27c29..25a1a5ba95d97e7da0459964464fb9b99a18badf)
- [1..2](https://github.com/hushsecurity/helm-charts/compare/2590ffca7cf52e4a9c017314a440eb2967f01661..3b430b9ddc3bfc91ed394ab653ef310738d27c29), [1..4](https://github.com/hushsecurity/helm-charts/compare/2590ffca7cf52e4a9c017314a440eb2967f01661..25a1a5ba95d97e7da0459964464fb9b99a18badf)
- [0..1](https://github.com/hushsecurity/helm-charts/compare/0d4d38026f6abf6f44a1d5885b5e7a13cc657683..2590ffca7cf52e4a9c017314a440eb2967f01661), [0..4](https://github.com/hushsecurity/helm-charts/compare/0d4d38026f6abf6f44a1d5885b5e7a13cc657683..25a1a5ba95d97e7da0459964464fb9b99a18badf)



[//]: # (gh-pr-revision:{"number":4,"hash":"25a1a5ba95d97e7da0459964464fb9b99a18badf","baseHash":"0d4d38026f6abf6f44a1d5885b5e7a13cc657683"})

**Review by horomena — APPROVED** [review]

**Review by sladkani — APPROVED** [review]

**Pull request merged**